### PR TITLE
Add placeholder `OpHardfork::Isthmus`

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/isthmus/overview.md

Add `OpHardfork::Isthmus`, without adding literal timestamps (ref https://github.com/paradigmxyz/reth/issues/13114).

### CODE REVIEW

**Comment by emhane** [comment]
blocked until next `op-alloy` release https://github.com/alloy-rs/op-alloy/pull/331

**Review by onbjerg — APPROVED** [review]
This was just released @emhane 🎊  Lgtm once bumped

**Review by mattsse — APPROVED** [review]
lgtm,

the todo!s are appropriate until relevant

**Pull request merged**
